### PR TITLE
workflows: bump sigstore-conformance

### PR DESCRIPTION
**Pull request by woodruffw**
Now with no labelling required!

### CODE REVIEW

File: .github/workflows/conformance.yml [hunk 1]
```diff
@@ -6,7 +6,6 @@ on:
       - main
   workflow_dispatch:
   pull_request_target:
```
Comment (woodruffw) [inline .github/workflows/conformance.yml:8]: We can get rid of `pull_request_target` entirely, right?

**Comment by tetsuo-cpp** [comment]
I think the labelling fix won't work until it's in `main`. But this should be good to go.

File: .github/workflows/conformance.yml (hunk 1 above)
Comment (woodruffw) [reply .github/workflows/conformance.yml:8]: (i.e. it should just be `pull_request`)

Comment (tetsuo-cpp) [reply .github/workflows/conformance.yml:8]: I think we can make it just `pull_request`.

Comment (tetsuo-cpp) [reply .github/workflows/conformance.yml]: [d735a8d](https://github.com/sigstore/sigstore-python/pull/637/commits/d735a8d2b632c63da610aac3f80b9ffa8e9df6b8)

**Review by tetsuo-cpp — APPROVED** [review]
LGTM!

**Pull request merged**